seaborn/rcmod.py

```diff
@@ -460,6 +433,35 @@ def set_context(context=None, font_scale=1, rc=None):
     mpl.rcParams.update(context_object)
 
 
+class _StyleOrContext(dict):
+    def __enter__(self):
+        rc = mpl.rcParams
+        self._orig = {k: rc[k] for k in self._keys}
+        type(self)._set(self)
```

Comment:
This seems overly complicated. Why not just `self._set()`? If you want, you can add a `_set` method to the base class that raises `NotImplementedError` to make it clear that derived classes need it. But I'm not aware of any reason you'd need to use `type` here.